crate/server/src/middlewares/jwt_auth.rs

```diff
@@ -130,11 +140,21 @@ where
 
         match private_claim {
             Err(_) => Box::pin(async move {
-                jwt_config.jwks.refresh().await?;
-
-                let private_claim = jwt_config
-                    .decode_bearer_header(&identity)
-                    .map(|claim| claim.email);
+                // Refresh jwks
+                jwt_config[0].jwks.refresh().await?;
+
+                for idp_config in jwt_config.iter() {
```

Comment:
If I am correct, this is essentially a copy/paste, from the above loop: make a dedicated primitive.

Reply:
After discussing it with @Manuthor  - I opened an issue for this : https://github.com/Cosmian/kms/issues/230